### PR TITLE
PFDR-38: validate bag according to Bagger JSON profile

### PR DESCRIPTION
**Pull request by aelkiss**
See https://github.com/LibraryOfCongress/bagger

### CODE REVIEW

File: spec/bagger_tag_spec.rb [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+          "fieldRequired": true,
+          "valueList": [
+            "allowed_value",
+            "another_allowed_value"
+          ]
+        }
+      }
+JSON
+
+  end
```
Comment (botimer) [inline spec/bagger_tag_spec.rb]: Some indentation issues here. Also, I've never seen the (<<XYZ) format -- neat. A little odd to have it match the class name, though.

File: spec/bagger_tag_spec.rb [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+    def expect_error_with_value(value,error_pattern)
+      subject.value_valid?(value,errors: errors)
+      expect(errors).to include a_string_matching error_pattern
+    end
+
+    context "with a required tag with no value list" do
+      subject { BaggerTag.new(name: "SomeField", required: true) }
+
+      it "is false when not given a value" do
```
Comment (botimer) [inline spec/bagger_tag_spec.rb]: The topic and conditions for these `it` statements are in the describe/context and should just be "is false" and "reports an error".

Maybe change the context to "with a required tag but no value list".

Likewise where applicable for the following examples.

File: spec/chipmunk_bag_validator_spec.rb [hunk 3]
```diff
@@ -25,8 +25,23 @@
...
+  let(:ext_validation_result) { ["", "", 0] }
+  let(:bag_info) { { "Foo" => "bar", "Baz" => "quux" } }
+  let(:tag_files) { good_tag_files }
+  let(:chipmunk_info) { chipmunk_info_good }
+
   let(:errors) { [] }
 
+  around(:each) do |example|
+    old_profile = Rails.application.config.validation["bagger_profile"]
+    profile_path = File.join(Rails.root,"spec","support","fixtures","test-profile.json")
```
Comment (botimer) [inline spec/chipmunk_bag_validator_spec.rb]: `Rails.root.join`

File: spec/bagger_tag_spec.rb (hunk 1 above)
Comment (aelkiss) [reply spec/bagger_tag_spec.rb]: The rubocop style guide requires using meaningful names for the heredoc delimiters. This also enables proper syntax highlighting with editor support.

I'll fix the indentation issues.

File: spec/bagger_tag_spec.rb (hunk 2 above)
Comment (aelkiss) [reply spec/bagger_tag_spec.rb]: The topic & conditions for the `it` statements are included in the context above (https://github.com/mlibrary/chipmunk/pull/33/files/b5762fc76e9e51ee40994d19690545c99d39dac3#diff-47c52f9c34a7dc40409bf0931d69b377R15) but I don't believe it's the case here. I agree the context names here could be approved. The context here is about the tag specification, the `it` is about the expected return & error when checking a given value against the tag specification.

**Pull request merged**